clab/docker.go

```diff
@@ -179,6 +187,52 @@ func (c *cLab) CreateContainer(ctx context.Context, node *Node) (err error) {
 	return createContainerNS(node.Pid, node.LongName)
 }
 
+func (c *cLab) PullImageIfRequired(ctx context.Context, node *Node) (err error) {
+	filter := filters.NewArgs()
+	filter.Add("reference", node.Image)
+
+	ilo := types.ImageListOptions{
+		All:     false,
+		Filters: filter,
+	}
+
+	log.Debugf("Looking up %s Docker image", node.Image)
+
+	images, err := c.DockerClient.ImageList(ctx, ilo)
+	if err != nil {
+		return err
+	}
+
+	// Image doesn't exist, need to pull it
+	if len(images) == 0 {
```

Comment:
lets keep the "happy path" on the first indentation level, meaning that the structure of the code should be revered like that:

```go
if len(images) > 0:
    log.Debugf("Image %s present, skip pulling", node.Image)
    return nil

// here goes the pulling logic on the same indentation level
```

Reply:
I dislike this approach.
The happy path in my view is when the last statement of a function is reached without any early exiting via any return other then the last line of a function... (exceptions exist of course)
However I'll change order and have the image exists atop.

Reply:
that is what is considered an idiomatic Go (left aligned happy path with early returns), so its good to adhere to what community adopted, that will make code look consistent

Reply:
Talking about idiomatic go: https://go-proverbs.github.io/
Last one is "don't panic" includes both panic statement and log.Fatalx

Reply:
> that is what is considered an idiomatic Go (left aligned happy path with early returns), so its good to adhere to what community adopted, that will make code look consistent

Well if that's the case ... who am I to complain, right?
It's changed!